src/os.cpp

```diff
@@ -999,3 +1000,266 @@ void os_stderr_set_color(TermColor color) {
     set_color_posix(color);
 #endif
 }
+
+static Buf* WindowsSDKPath = buf_alloc();
```

Comment:
No code execution in static initializers please. Two reasons: 1, we don't have code execution in static initializers in zig, and the C++ code is roughly written in a zig style even though it's c++. 2. code execution in static initializers is kind of a hack, that I'd rather not depend on. Better to not have too much hidden stuff run before main().

More importantly, I think that os.cpp is not a good place to save state such as WindowsSDKPath and WindowsSDKVersion. For example, when zig acts as a compiler server (which is planned) we don't want to have to restart Zig if you install/uninstall MSVC with zig running.

A better place to save the state of already having found MSVC is the CodeGen struct (in all_types.hpp).

It's still good to isolate the COM stuff into this os.cpp file. But os.hpp could have an opaque type that has the found MSVC state which codegen/analyze can pass to e.g. `os_get_win32_ucrt_lib_path`. And then we don't want to forget to pass the state from parent CodeGen to child CodeGen in link.cpp (in the code where it builds compiler_rt.o and builtin.o).

If this is unclear, let me know, I'd be happy to finish that up in this pull request. You've done the bulk of the work, and what's left is kind of the glue / refactoring which I'd be happy to take care of for you.